cpp/arcticdb/storage/s3/detail-inl.hpp

```diff
@@ -44,50 +44,47 @@ namespace s3 {
         template<class It>
         using Range = folly::Range<It>;
 
-        class UnexpectedS3ErrorException : public std::exception {
-        public:
-            UnexpectedS3ErrorException(const std::string& message):message(message){}
-            const char* what() const noexcept override {
-                return message.c_str();
+        inline void handle_s3_error(const Aws::S3::S3Error& err){
```

Comment:
I agree with the `inline` point. The function has to be in the header file as it is called by the functions declared in this file. Should we make the `detail-inl.hpp` file a source file rather than a header?

Reply:
On second thought let's leave it as is

I think we should do a separate refactor to get rid of this -inl.hpp file completely

But it will make the PR too hard to review if we do that right now